ui/main.slint

```diff
@@ -0,0 +1,106 @@
+import { MessageScreen } from "screens/message.slint";
+
+export component AppWindow inherits Window {
+    property <[string]> button_str: ["Connect", "Disconnect", "Disconnecting...", "Connecting..."];
+    property <[string]> status_str: ["Disconnected", "Connected", "Disconnecting", "Connecting"];
+
+    in property <string> window_title;
+    in property <int> status;
+    in property <bool> freeze;
+    in property <[LineEditInfo]> profiles <=> profiles_screen.info;
+    in-out property <string> message;
+    in-out property <[LineEditInfo]> manual: [
+        { label: "Address:", text: "" },
+        { label: "Port:", text: "" },
+        { label: "Root certificate:", text: "" },
+        { label: "Certificate:", text: "" },
+        { label: "Key:", text: "" },
+    ];
+    in-out property <LineEditInfo> from_file: { label: "", text: "" };
+    out property <int> menu_index: 0;
+
+    callback browse <=> add_profile_screen.browse;
+    callback reload;
+    callback disconnect();
+    callback connect();
+
+    preferred-width: 1280px;
+    preferred-height: 720px;
+    min-width: 640px;
+    min-height: 360px;
+    title: window_title;
+
+    blur := Blur {
+        z: 2;
+        trigger: parent.freeze;
+        freeze: true;
+    }
+
+    message_screen := MessageScreen {
+        z: 10;
+        text <=> parent.message;
+    }
+
+    top_panel := TopPanel {
+        width: parent.width;
+        text-left: menu_screen.section;
+        text-right: parent.status_str[status] + (parent.status == 1 ? " to " + parent.from_file.label : "");
+    }
+
+    menu_screen := MenuScreen {
+        z: 1;
+        clicked(i) => {
+            if (i != parent.menu_index) {
+                parent.from_file = { label: "", text: "" };
+                profiles_screen.selected = -1;
+                add_profile_screen.selection = 0;
+            }
+            parent.menu_index = i;
+        }
+    }
+
+    add_profile_screen := AddProfileScreen {
+        y: top_panel.height - parent.height * 1.5 * (parent.menu_index - 0);
+        height: parent.height - top_panel.height - bottom_button.height;
+        manual_info <=> parent.manual;
+        from_file_info <=> parent.from_file;
+    }
+
+    profiles_screen := ProfilesScreen {
+        y: top_panel.height - parent.height * 1.5 * (parent.menu_index - 1);
+        height: parent.height - top_panel.height - bottom_button.height;
+        clicked(info) => {
+            parent.from_file = info;
+        }
+
+        reload => {
+            parent.reload();
+            if (self.selected != -1 && self.selected < parent.profiles.length) {
+                parent.from_file = profiles[self.selected];
+            } else {
+                parent.from_file = { label: "", text: "" };
+                self.selected = -1;
+            }
+        }
+    }
+
+    bottom_button := BottomButton {
+        touch_area-visible: parent.from_file.text != "" && parent.status < 2;
```

Comment:
`2` looks like some magic number here. Maybe some named constants and more readable code?